docs/portal.properties-Reference.md

```diff
@@ -280,3 +281,38 @@ This gene set will add the following in the query box:
 ```
 "BRCA genes" BRCA1: MUT=E1258D; BRCA2: HOMDEL MUT=NONSENSE MUT=NONSTART MUT=NONSTOP MUT=FRAMESHIFT MUT=SPLICE MUT=TRUNC;
 ```
+# EHCache Settings
+cBioPortal is supported on the backend with EHCache. The default setting initializes two seperate hybrid (disk + heap) caches. The configuration, size, and location of these caches are configurable from within portal.properties through the following properties.
+
+First, select a cache configuration using `ehcache.xml.configuration`. This specifies whether to use a disk-only, heap-only, or hybrid cache. The default value is `ehcache-mixed.xml` which initializes a hybrid caching system.
+```
+ehcache.xml.configuration=[ehcache-heap-only.xml OR ehcache-mixed.xml or ehcache-disk-only.xml]
+```
+
+Cache location is set by setting `ehcache.persistence.path` to a existing path on the filesystem. EHCache will create seperate directories under the provided path for each cache defined in the ehcache.xml configuration file. 
+```
+ehcache.persistence.path=[location on the disk filesystem where EHCache can write the cache to /tmp/]
+```
+
+Cache size must be set for both heap and disk; EHCache requires disk size to be greater than heap size. Default values are provided. The general repository cache is specified to use 1021MB for heap and 4GB for disk. The static repository cache is specified to use 30MB for heap and 32MB for disk. For installations with increased traffic or data, cache sizes can be increased to further improve performance. 
+```
+ehcache.general_repository_cache.max_bytes_heap=
+ehcache.general_repository_cache.max_bytes_heap_units=[size unit e.g MB, GB, entries]
+ehcache.general_repository_cache.max_bytes_local_disk=
+ehcache.general_repository_cache.max_bytes_local_disk_units=[size unit e.g MB, GB, entries]
+
+ehcache.static_repository_cache_one.max_bytes_heap=
+ehcache.static_repository_cache_one.max_bytes_heap_units=[size unit e.g MB, GB, entries]
+ehcache.static_repository_cache_one.max_bytes_local_disk=
+ehcache.static_repository_cache_one.max_bytes_local_disk_units[size unit e.g MB, GB, entries]
+```
+
+Additional properties can be specified for cache statistics monitoring. To log metrics regarding memory usage, set `ehcache.enable.statistics` to true. Logged metrics and additional information such as cache size and cached keys are available through an optional endpoint. The optional endpoint is turend off by default but can turned on by setting `cache.statistics.endpoint.enabled` to true. 
```

Comment:
I think we should add a warning that the cache statistics endpoint will expose database query parameters, such as sample and patient identifiers, study names, and other information which is normally is not visible to users unless they are granted authorities to access certain studies. If the deployer is deploying a protected portal where users have authorities to see only a subset of the data from the database, they should think carefully before enabling this endpoint.

Also, the cache statistics endpoint will not show up in the swagger page documentation. If we tell the deployers here that they can enable it, we should also show examples of how the endpoints can be accessed when the endpoint is enabled (so they can test that it is working).